WalletWasabi/Services/CcjCoordinator.cs

```diff
@@ -294,20 +294,27 @@ public bool ContainsCoinJoin(uint256 hash)
 				}
 				else
 				{
-					foreach (var cjHash in UnconfirmedCoinJoins)
+					foreach (var cjHash in UnconfirmedCoinJoins.ToArray())
 					{
-						RPCResponse getRawTransactionResponse = await RpcClient.SendCommandAsync(RPCOperations.getrawtransaction, cjHash.ToString(), true);
-						// if failed remove from everywhere (should not happen normally)
-						if (string.IsNullOrWhiteSpace(getRawTransactionResponse?.ResultString))
+						try
 						{
-							UnconfirmedCoinJoins.Remove(cjHash);
-							CoinJoins.Remove(cjHash);
-							await File.WriteAllLinesAsync(CoinJoinsFilePath, CoinJoins.Select(x => x.ToString()));
+							var txInfo = await RpcClient.GetRawTransactionInfoAsync(cjHash);
+
+							// if confirmed remove only from unconfirmed
+							if (txInfo.Confirmations > 0)
+							{
+									UnconfirmedCoinJoins.Remove(cjHash);
+							}
 						}
-						// if confirmed remove only from unconfirmed
-						if (getRawTransactionResponse.Result.Value<int>("confirmations") > 0)
+						catch(RPCException e)
 						{
+							if(e.RPCCode != RPCErrorCode.RPC_INVALID_ADDRESS_OR_KEY)
```

Comment:
Why do you throw?

Reply:
Because if there is another kind of RPCException, whatever it could be, I want to know it and I don't want to perform changes in the Confirmed/Unconfirmed coinjoin collections simply because there was an connection problem, the node is not ready yet or any other unexpected situation. My general rule in this cases is never swallow exceptions.

Reply:
But this worsens the reliability of the server. If an exception is thrown consistently for whatever reason, then it'll make everyone unable to register inputs. It'd be better to log it and autocorrect.

Reply:
`bitcoind` will always be ready, it's asserted when our API initializes.